### PR TITLE
API-8350 refactor only

### PR DESCRIPTION
**Pull request by crolarlibertyva**
- This just refactors the method
- There will be a followup PR that includes the jest upgrade and the jest test changes

### CODE REVIEW

File: src/routes/handlers.js [hunk 1]
```diff
@@ -173,3 +158,58 @@ export const handleError = (req, res) => {
   logger.error({ idp_sid: req.cookies.idp_sid });
   res.render(urlUserErrorTemplate(req), { request_id: rTracer.id() });
 };
+
+const getSamlRequestUrl = (options, params, exParams) => {
```
Comment (beaugrantham-va) [inline src/routes/handlers.js:162]: This needs to be documented. I have no idea what sort of options, params, or exParams I can or must provide.

Comment (crolarlibertyva) [reply src/routes/handlers.js:162]: Will do the same for other broken out functions.

File: src/routes/handlers.js [hunk 2]
```diff
@@ -173,3 +158,58 @@ export const handleError = (req, res) => {
...
+          .then((url) => {
+            if (url) {
+              authOpts[key] = url;
+              resolve(authOpts);
+            }
+          })
+          .catch((err) => reject(err));
+      });
+    })
+    .catch(next);
```
Comment (beaugrantham-va) [inline src/routes/handlers.js:214]: I'm not sure this error handling is functionally equivalent. It's hard to follow the chaining, but it looks like if a promise is rejected that this will hide it from the caller... who is also trying to `catch(next)`?

I'm not comfortable with these changes merging without accompanying unit level tests which cover error handling scenarios.
